packages/cli/src/commands/db/revert.ts

```diff
@@ -9,6 +9,38 @@ import type { Migration } from '@db/types';
 import { wrapMigration } from '@db/utils/migrationHelpers';
 import config from '@/config';
 
+// This function is extracted to make it easier to unit test it.
+// Mocking turned into a mess due to this command using typeorm and the db
+// config directly and customizing and monkey patching parts.
+export async function main(
```

Comment:
If you have a better idea for unit testing, let me know. But everything I tried was messy due to `jest.mock` not working nicely with typeorm, because it will mock the decorators which then stop working and `jest-extended-mock` will mock based on the type that is used to to retrieve the object, not the actual implementation, and thus will actually add methods which I rely on not existing, like `Migration.down`.

Using DI meant changing `Db.init` to take custom data source options and eventually led to the test being in an infinite loop that I didn't investigate for long.

So this is where I ended up for now.